Build failure: ceph

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

(initially reported as a comment in https://github.com/NixOS/nixpkgs/issues/369777)

`nix build .#ceph`



### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/lmnvsg49akzmvwj417zgm7gk5qnzs458-ceph-19.2.0.drv' failed with exit code 2;
       last 25 log lines:
       > include/boost/asio/io_context.hpp:936:42: warning: '((const boost::asio::io_context::basic_executor_type<std::allocator<void>, 4>*)this)[5].boost::asio::io_context::basic_executor_type<std::allocator<void>, 4>::target_' may be used uninitialized [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized8;;]
       > In member function 'boost::asio::io_context* boost::asio::io_context::basic_executor_type<Allocator, Bits>::context_ptr() const [with Allocator = std::allocator<void>; long unsigned int Bits = 4]',
       >     inlined from 'boost::asio::io_context::basic_executor_type<Allocator, Bits>::~basic_executor_type() [with Allocator = std::allocator<void>; long unsigned int Bits = 4]' at include/boost/asio/io_context.hpp:523:22,
       >     inlined from 'boost::asio::executor_work_guard<Executor, typename std::enable_if<(! boost::asio::is_executor<T>::value), void>::type, typename std::enable_if<boost::asio::execution::is_executor<T>::value, void>::type>::~executor_work_guard() [with Executor = boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>]' at include/boost/asio/executor_work_guard.hpp:202:70,
       >     inlined from 'std::pair<boost::asio::executor_work_guard<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, void, void>, boost::asio::executor_work_guard<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, void, void> >::~pair()' at /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/stl_pair.h:284:12,
       >     inlined from 'virtual ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Linger_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>::~CompletionImpl()' at /build/ceph-19.2.0/src/common/async/completion.h:166:7,
       >     inlined from 'constexpr void std::destroy_at(_Tp*) [with _Tp = ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<allocator<void>, 0>, Objecter::CB_Linger_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>]' at /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/stl_construct.h:88:18,
       >     inlined from 'static constexpr void std::allocator_traits<std::allocator<_Up> >::destroy(allocator_type&, _Up*) [with _Up = ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Linger_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>; _Tp = ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Linger_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>]' at /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/alloc_traits.h:593:19,
       >     inlined from 'void ceph::async::detail::CompletionImpl<Executor1, Handler, T, Args>::destroy() [with Executor1 = boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>; Handler = Objecter::CB_Linger_Map_Latest; T = void; Args = {boost::system::error_code, long unsigned int, long unsigned int}]' at /build/ceph-19.2.0/src/common/async/completion.h:225:27,
       >     inlined from 'static void ceph::async::detail::CompletionImpl<Executor1, Handler, T, Args>::operator delete(void*) [with Executor1 = boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>; Handler = Objecter::CB_Linger_Map_Latest; T = void; Args = {boost::system::error_code, long unsigned int, long unsigned int}]' at /build/ceph-19.2.0/src/common/async/completion.h:248:45,
       >     inlined from 'virtual ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Linger_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>::~CompletionImpl()' at /build/ceph-19.2.0/src/common/async/completion.h:166:7:
       > include/boost/asio/io_context.hpp:936:42: warning: '((const boost::asio::io_context::basic_executor_type<std::allocator<void>, 4>*)this)[5].boost::asio::io_context::basic_executor_type<std::allocator<void>, 4>::target_' may be used uninitialized [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized8;;]
       > In member function 'boost::asio::io_context* boost::asio::io_context::basic_executor_type<Allocator, Bits>::context_ptr() const [with Allocator = std::allocator<void>; long unsigned int Bits = 4]',
       >     inlined from 'boost::asio::io_context::basic_executor_type<Allocator, Bits>::~basic_executor_type() [with Allocator = std::allocator<void>; long unsigned int Bits = 4]' at include/boost/asio/io_context.hpp:523:22,
       >     inlined from 'boost::asio::executor_work_guard<Executor, typename std::enable_if<(! boost::asio::is_executor<T>::value), void>::type, typename std::enable_if<boost::asio::execution::is_executor<T>::value, void>::type>::~executor_work_guard() [with Executor = boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>]' at include/boost/asio/executor_work_guard.hpp:202:70,
       >     inlined from 'std::pair<boost::asio::executor_work_guard<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, void, void>, boost::asio::executor_work_guard<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, void, void> >::~pair()' at /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/stl_pair.h:284:12,
       >     inlined from 'virtual ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Command_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>::~CompletionImpl()' at /build/ceph-19.2.0/src/common/async/completion.h:166:7,
       >     inlined from 'constexpr void std::destroy_at(_Tp*) [with _Tp = ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<allocator<void>, 0>, Objecter::CB_Command_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>]' at /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/stl_construct.h:88:18,
       >     inlined from 'static constexpr void std::allocator_traits<std::allocator<_Up> >::destroy(allocator_type&, _Up*) [with _Up = ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Command_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>; _Tp = ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Command_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>]' at /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/alloc_traits.h:593:19,
       >     inlined from 'void ceph::async::detail::CompletionImpl<Executor1, Handler, T, Args>::destroy() [with Executor1 = boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>; Handler = Objecter::CB_Command_Map_Latest; T = void; Args = {boost::system::error_code, long unsigned int, long unsigned int}]' at /build/ceph-19.2.0/src/common/async/completion.h:225:27,
       >     inlined from 'static void ceph::async::detail::CompletionImpl<Executor1, Handler, T, Args>::operator delete(void*) [with Executor1 = boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>; Handler = Objecter::CB_Command_Map_Latest; T = void; Args = {boost::system::error_code, long unsigned int, long unsigned int}]' at /build/ceph-19.2.0/src/common/async/completion.h:248:45,
       >     inlined from 'virtual ceph::async::detail::CompletionImpl<boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>, Objecter::CB_Command_Map_Latest, void, boost::system::error_code, long unsigned int, long unsigned int>::~CompletionImpl()' at /build/ceph-19.2.0/src/common/async/completion.h:166:7:
       > include/boost/asio/io_context.hpp:936:42: warning: '((const boost::asio::io_context::basic_executor_type<std::allocator<void>, 4>*)this)[5].boost::asio::io_context::basic_executor_type<std::allocator<void>, 4>::target_' may be used uninitialized [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized8;;]
       > [ 34%] Built target common-objs
       > make: *** [Makefile:146: all] Error 2
       For full logs, run 'nix log /nix/store/lmnvsg49akzmvwj417zgm7gk5qnzs458-ceph-19.2.0.drv'.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.0, NixOS, 25.05 (Warbler), 25.05.20250121.9e4d519`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/irknf7niks2djxz4csh7sr44wwrs082y-source`

### Notify maintainers

@adevress
@alexanderkjeldaas
@johanot
@krav
@nh2

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/filesystems/ceph/default.nix
+++ b/pkgs/tools/filesystems/ceph/default.nix
@@ -26,7 +26,7 @@
   # Runtime dependencies
   arrow-cpp,
   babeltrace,
-  boost,
+  boost186,
   bzip2,
   cryptsetup,
   cunit,
@@ -289,7 +289,7 @@ let
       };
   };
 
-  boost' = boost.override {
+  boost' = boost186.override {
     enablePython = true;
     inherit python;
   };

